pkg/dao/templates.go

```diff
@@ -167,6 +167,32 @@ func (t templateDaoImpl) filteredDbForList(orgID string, filteredDB *gorm.DB, fi
 	return filteredDB
 }
 
+func (t templateDaoImpl) SoftDelete(orgID string, uuid string) error {
+	var modelTemplate models.Template
+	var respTemplate api.TemplateResponse
+
+	err := t.db.Where("uuid = ? AND org_id = ?", UuidifyString(uuid), orgID).First(&modelTemplate).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return &ce.DaoError{NotFound: true, Message: "Could not find template with UUID " + uuid}
+		}
+		return t.DBToApiError(err)
+	}
+
+	if err = t.db.Delete(&modelTemplate).Error; err != nil {
+		return err
+	}
+
+	templatesModelToApi(modelTemplate, &respTemplate)
```

Comment:
the response template is not needed, because this doesn't return a response, right?

Reply:
lol yes, fixed :)